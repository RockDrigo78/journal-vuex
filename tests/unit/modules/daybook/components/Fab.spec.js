import { shallowMount } from '@vue/test-utils';
import Fab from '@/modules/daybook/components/Fab.vue';

describe('Fab component tests', () => {
    test('should match snapshot', () => {
        const wrapper = shallowMount(Fab);
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('should show default icon', () => {
        const wrapper = shallowMount(Fab);
        expect(wrapper.find('i').classes()).toContain('fa-plus');
    });

    test('should show sent icon', () => {
        const wrapper = shallowMount(Fab, {
            propsData: {
                icon: 'fa-save',
            },
        });

        expect(wrapper.props().icon).toBe('fa-save');
        expect(wrapper.find('i').classes()).toContain('fa-save');
    });

    test('should emmit on:click event on button click', () => {
        const wrapper = shallowMount(Fab);

        wrapper.find('button').trigger('click');

        expect(wrapper.emitted('on:click')).toHaveLength(1);
    });
});
