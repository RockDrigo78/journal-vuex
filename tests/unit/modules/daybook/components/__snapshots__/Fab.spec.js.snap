// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab component tests should match snapshot 1`] = `<button class="btn btn-primary"><i class="fa fa-2x fa-plus"></i></button>`;
